main/internal/implementations/dragon_drop/http_dragondrop_managed_visualization_test.go

```diff
@@ -0,0 +1,808 @@
+			WorkspaceDirectories: []string{"workspace1"},
+		},
+	}
+
+	err := writeFile("another.tf", []byte(`
+		terraform {
+		  required_version = "1.5.1"
+		
+		  required_providers {
+			google = {
+			  source  = "hashicorp/google"
+			  version = "~>4.77.0"
+			}
+		  }
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	// When
+	terraformFootprint, err := client.getTerraformFootprintData(ctx)
+
+	// Then
+	require.NoError(t, err)
+	require.Equal(t, `{"4.77.0":1}`, terraformFootprint.VersionsTFProviders)
+	require.Equal(t, `{"1.5.1":1}`, terraformFootprint.VersionsTF)
+	require.Equal(t, "{}", terraformFootprint.VersionsTFModules)
+}
+
+func TestHTTPDragonDropClient_getTerraformFootprintData_another_tf_file_one_module(t *testing.T) {
+	// Given
+	defer cleanMockedDirectory()
+
+	ctx := context.Background()
+	client := HTTPDragonDropClient{
+		config: HTTPDragonDropClientConfig{
+			WorkspaceDirectories: []string{"workspace1"},
+		},
+	}
+
+	err := writeFile("another.tf", []byte(`
+		terraform {
+		  required_version = "1.5.1"
+		
+		  required_providers {
+			google = {
+			  source  = "hashicorp/google"
+			  version = "~>4.77.0"
+			}
+		  }
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file.tf", []byte(`
+		module "servers" {
+			source = "./app-cluster"
+			version = "0.0.5"
+		
+			servers = 5
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	// When
+	terraformFootprint, err := client.getTerraformFootprintData(ctx)
+
+	// Then
+	require.NoError(t, err)
+	require.Equal(t, `{"4.77.0":1}`, terraformFootprint.VersionsTFProviders)
+	require.Equal(t, `{"1.5.1":1}`, terraformFootprint.VersionsTF)
+	require.Equal(t, `{"0.0.5":1}`, terraformFootprint.VersionsTFModules)
+}
+
+func TestHTTPDragonDropClient_getTerraformFootprintData_another_tf_file_two_modules_in_a_file(t *testing.T) {
+	// Given
+	defer cleanMockedDirectory()
+
+	ctx := context.Background()
+	client := HTTPDragonDropClient{
+		config: HTTPDragonDropClientConfig{
+			WorkspaceDirectories: []string{"workspace1"},
+		},
+	}
+
+	err := writeFile("another.tf", []byte(`
+		terraform {
+		  required_version = "1.5.1"
+		
+		  required_providers {
+			google = {
+			  source  = "hashicorp/google"
+			  version = "~>4.77.0"
+			}
+		  }
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file.tf", []byte(`
+		module "servers" {
+			source = "./app-cluster"
+			version = "0.0.5"
+		
+			servers = 5
+		}
+
+		module "servers2" {
+			source = "./app-cluster"
+			version = "0.0.6"
+		
+			servers = 5
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	// When
+	terraformFootprint, err := client.getTerraformFootprintData(ctx)
+
+	// Then
+	require.NoError(t, err)
+	require.Equal(t, `{"4.77.0":1}`, terraformFootprint.VersionsTFProviders)
+	require.Equal(t, `{"1.5.1":1}`, terraformFootprint.VersionsTF)
+	require.Equal(t, `{"0.0.5":1,"0.0.6":1}`, terraformFootprint.VersionsTFModules)
```

Comment:
```suggestion
	require.Equal(t, `'{./app-cluster':{"0.0.5":1,"0.0.6":1}}`, terraformFootprint.VersionsTFModules)
```
We need to capture the versions by module type here otherwise we cannot filter by module type in the front end: https://docs.google.com/spreadsheets/d/18aZpnUSbvEwg1du1bYr5mRT6IiqbTzU25gULKwD09OY/edit#gid=1016560229